include/TextInputWidget.h

```diff
@@ -29,26 +31,36 @@
 
 namespace msgs = matrix::events::messages;
 
-static const QString EMOTE_COMMAND("/me ");
-static const QString JOIN_COMMAND("/join ");
-
 class FilteredTextEdit : public QTextEdit
 {
         Q_OBJECT
 
-private:
-        QTimer *typingTimer_;
-
 public:
         explicit FilteredTextEdit(QWidget *parent = nullptr);
-        void keyPressEvent(QKeyEvent *event);
 
         void stopTyping();
 
+        QSize sizeHint() const override;
+        QSize minimumSizeHint() const override;
+
+        void submit();
+
 signals:
-        void enterPressed();
         void startedTyping();
         void stoppedTyping();
+        void message(QString);
+        void command(QString name, QString args);
+
+protected:
+        void keyPressEvent(QKeyEvent *event) override;
+
+private:
+        std::deque<QString> true_history_, working_history_;
```

Comment:
Why not use a `QQueue`? It doesn't really matter, but as far as I can see, the rest of the codebase uses mostly Qt containers, not STL ones.

Reply:
Why use a nonstandard container when a standard one will do just fine?

Reply:
*C O N S I S T E N C Y* 😎 👉 👉

Reply:
Consistency with the rest of the world is among the advantages of standardization, yes :smile:

Reply:
I mean in the nheko codebase. If you search for `std::` in the whole source then you'll see that not a single container is used, mostly things like `std::max` or functions for randomization.

Using `std::queue` instead of `QQueue` is just a bit alien. As I said, it doesn't really matter, it's just different than everything else.